src/Loggable/LoggableListener.php

```diff
@@ -229,6 +242,39 @@ protected function getLogEntryClass(LoggableAdapter $ea, $class)
         return self::$configurations[$this->name][$class]['logEntryClass'] ?? $ea->getDefaultLogEntryClass();
     }
 
+    /**
+     * Retrieve the username to use for the log entry.
+     *
+     * This method will try to fetch a username from the actor provider first, falling back to the {@see $this->username}
+     * property if the provider is not set or does not provide a value.
+     */
+    protected function getUsername(): ?string
+    {
+        if ($this->actorProvider instanceof ActorProviderInterface) {
+            $actor = $this->actorProvider->getActor();
+
+            if (is_string($actor) || null === $actor) {
+                return $actor;
+            }
+
+            if (method_exists($actor, 'getUserIdentifier')) {
+                return (string) $actor->getUserIdentifier();
+            }
+
+            if (method_exists($actor, 'getUsername')) {
+                return (string) $actor->getUsername();
+            }
+
+            if (method_exists($actor, '__toString')) {
+                return $actor->__toString();
+            }
+
+            throw new InvalidArgumentException(\sprintf('The loggable extension requires the actor provider to return a string or an object implementing the "getUserIdentifier", "getUsername", or "__toString" methods. "%s" cannot be used as an actor.', get_class($actor)));
```

Comment:
Since the actor is not injected as an argument here, I think `UnexpectedValueException` is better for this case.

Reply:
Updated.